experimental/generation/generator/packages/library/src/dialogGenerator.ts

```diff
@@ -236,17 +236,20 @@ function addPrefix(prefix: string, name: string): string {
     }
 }
 
-// Add entry to the .lg generation context and return it.  
+// Add information about a newly generated file.
 // This also ensures the file does not exist already.
-type FileRef = {name: string, fallbackName: string, fullName: string, relative: string}
-function addEntry(fullPath: string, outDir: string, tracker: any): FileRef | undefined {
+type FileRef = {name: string, shortName: string, fallbackName: string, fullName: string, relative: string}
+function addFileRef(fullPath: string, outDir: string, prefix: string, tracker: any): FileRef | undefined {
     let ref: FileRef | undefined
     let basename = ppath.basename(fullPath, '.dialog')
     let ext = ppath.extname(fullPath).substring(1)
     let arr: FileRef[] = tracker[ext]
     if (!arr.find(ref => ref.name === basename)) {
+        let shortName = basename.substring(prefix.length + 1, basename.indexOf('.'))
```

Comment:
Normally, I would have some length checking here.  Are you OK if shortName ends up being empty?